vendor/github.com/openshift/cluster-api/pkg/controller/machine/testactuator.go

```diff
@@ -91,6 +91,12 @@ func (a *TestActuator) Exists(context.Context, *v1alpha1.Cluster, *v1beta1.Machi
 	return a.ExistsValue, nil
 }
 
+func (a *TestActuator) GetInstanceTypeDetails(_ context.Context, _ *v1beta1.MachineSet) (map[string]string, error) {
```

Comment:
function description would help doing early review. Is this suppose to figure out machine from the machineset and then collect annotations from the machine and return those? If yes, what if machines belonging to a machineset have different annotations?